### PR TITLE
Properly inherit text decorations

### LINKED ISSUE #8684: non-inline block resets underline property
```html
<style>
.u { text-decoration: underline; }
</style>
<div class='u'>a</div>
<div class='u'><div>b</div></div>
<div class='u'><div style="display:inline">c</div></div>
```

-------

expected:

screen shot 2015-11-25 at 9 41 52 pm

-------

actual:

screen shot 2015-11-25 at 9 42 02 pm

---------

`tests/wpt/web-platform-tests/html/rendering/non-replaced-elements/phrasing-content-0/font-element-text-decoration-color/001-a.html` tests this behavior and is currently failing

### LINKED ISSUE #8783: text_decoration_propagation_a.html fails
Disabled in a7fcbd9d6035bfab183a0dabc38b053c2a080abf

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #8684.
Fixes #8783.

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10438)

### CODE REVIEW

**Comment by frewsxcv** [comment]
Can you add https://github.com/servo/servo/issues/8684 as a regression test?

**Comment by KiChjang** [comment]
Done.

File: tests/wpt/mozilla/tests/mozilla/non-inline-block-resets-underline-property-ref.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!doctype html>
+<meta charset="utf-8">
+<title></title>
+<link rel="match" href="non-inline-block-resets-underline-property-ref.html">
```
Comment (frewsxcv) [inline tests/wpt/mozilla/tests/mozilla/non-inline-block-resets-underline-property-ref.html]: This link is probably not necessary

**Comment by frewsxcv** [comment]
Sorry to do this to you, but maybe the tests are more appropriate if they lived in `tests/wpt/mozilla/tests/css/`

**Comment by KiChjang** [comment]
Done.

**Comment by jdm** [comment]
MANIFEST.json isn't quite right according to TravisCI.

**Comment by KiChjang** [comment]
Should be OK now. I get the feeling that we should write a script to fix this...

**Comment by jdm** [comment]
You mean `./mach test-wpt --manidest-update SKIP_TESTS`?

**Comment by KiChjang** [comment]
Wow, never knew about that. ./mach help doesn't tell me that it accepts a SKIP_TESTS parameter though.

r? @mbrubeck

**Comment by jdm** [comment]
`SKIP_TESTS` is the name of the tests to run, so no tests end up running since none match that name.

**Comment by mbrubeck** [comment]
@bors-servo r+

---

Reviewed 1 of 2 files at r1, 3 of 5 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, 1 of 1 files at r7.
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10438#-:-KEm2TGtlkUV7X-dkOvL:1236647940)*

**Comment by frewsxcv** [comment]
@bors-servo r-

```
Tests with unexpected results:
  ▶ PASS [expected FAIL] /css21_dev/html4/text-decoration-applies-to-006.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/text-decoration-applies-to-008.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/text-decoration-applies-to-011.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/text-decoration-applies-to-009.htm

  ▶ PASS [expected FAIL] /css21_dev/html4/text-decoration-applies-to-010.htm
```

**Comment by KiChjang** [comment]
@frewsxcv I already updated the test expectations and pushed.

@bors-servo r=mbrubeck

**Pull request merged**